src/main/java/com/box/sdk/BoxAPIRequest.java

```diff
@@ -404,6 +405,66 @@ public BoxAPIResponse send(ProgressListener listener) {
         throw new RuntimeException();
     }
 
+    /**
+      * Sends a request to upload a file part and returns a BoxFileUploadSessionPart containing information
+      * about the upload part. This method is separate from send() because it has custom retry logic.
+      *
+      * <p>If the server returns an error code or if a network error occurs, then the request will be automatically
+      * retried. If the maximum number of retries is reached and an error still occurs, then a {@link BoxAPIException}
+      * will be thrown.</p>
+      *
+      * @param session The BoxFileUploadSession uploading the part
+      * @param offset Offset of the part being uploaded
+      * @throws BoxAPIException if the server returns an error code or if a network error occurs.
+      * @return A {@link BoxFileUploadSessionPart} part that has been uploaded.
+      */
+    public BoxFileUploadSessionPart sendForUploadPart(BoxFileUploadSession session, long offset) {
```

Comment:
If we don't want to expose this method and it's only meant for internal package use only for chunked upload we should consider different modifier, maybe default or protected?